lib/pbench/agent/utils.py

```diff
@@ -188,3 +190,40 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+# From https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address
+_allowed = re.compile(
+    r"^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$"
+)
+
+
+def validate_hostname(host_name):
+    """validate_hostname - validate the given hostname uses the proper syntax.
+
+    A host name that follows RFC 952 (amended by RFC 1123) is accepted only.  Host
+    names are not resolved to IP addresses, and IP addresses are also accepted.
+
+    Algorithm taken from: https://stackoverflow.com/questions/2532053/validate-a-hostname-string
+
+    Returns 0 on success, 1 on failure.
+    """
+    if len(host_name) > 255:
+        return 1
+
+    if host_name[-1] == ".":
+        # Strip trailing '.' to allow comprehension below to work in all cases
+        _host_name = host_name[:-1]
+    else:
+        _host_name = host_name
+    if all(_allowed.match(x) for x in _host_name.split(".")):
+        return 0
+
+    # It is not a valid host name, but could be a valid IP address.
+    try:
+        ipaddress.ip_address(host_name)
+    except ValueError:
+        ret_val = 1
+    else:
+        ret_val = 0
+    return ret_val
```

Comment:
If you accept my suggested RE above, this code should never be presented with any valid IPv4 addresses.  However, it does validate IPv6 ones, right?  It would be worth updating the code comment to that effect.

Reply:
I'll address in PR #2207.